app/controllers/ResultController.scala

```diff
@@ -84,33 +84,33 @@ trait ResultController extends BaseController with AuthFunction {
 
 
   private def routeViaMCNeededCheck(response: HttpResponse)(implicit request: Request[AnyContent], protectionType: ApplicationType.Value): Future[Result] = {
-    response.status match {
-      case 423 => Future.successful(Locked(manualCorrespondenceNeeded()))
-      case _ => saveAndRedirectToDisplay(response)
+    genericAuthWithNino("existingProtections") { nino =>
```

Comment:
This should be moved up a level and nino should be passed into routeViaMCCheckNeeded